src/NUnitFramework/framework/Constraints/Comparers/StreamsComparer.cs

```diff
@@ -82,7 +82,6 @@ public static EqualMethodResult Equal(object x, object y, ref Tolerance toleranc
                             return EqualMethodResult.ComparedNotEqual;
                         }
                     }
-                    readByte += BUFFER_SIZE;
```

Comment:
Great find on this not being needed!

One thing you can try for a test is to have the tests define a new class which extends `MemoryStream` and overrides `Read(byte[] buffer, int offset, int count)` method so that it calls `base.Read(buffer, offset, 2)` to simulate a steam which always returns less than the asked for amount. Your test can then instantiate two copies of this class, one for _expected_ and one for _actual_, each of which can be given a small byte array of equal length but with a differing value somewhere near the end of the array.

```csharp
private class ShortReadingMemoryStream : MemoryStream
{
    public ShortReadingMemoryStream(byte[] bytes) : base(bytes) { }

    public override int Read(byte[] buffer, int offset, int count)
    {
        return base.Read(buffer, offset, 2);
    }
}
```

We may also want to change line 71 above to end use `readActual` instead of `BUFFER_SIZE` in the `for` loop declaration to avoid reading past the part of the buffer filled by the `Read()`.

Reply:
Actually, while doing more investigation around this change, Linq has a [SequenceEquals ](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.sequenceequal?view=net-8.0) which gives the same performance enhancement as my previous PR whilst removing one of the dependancies I added in the previous PR (System.Memory).

I've also pushed the changes in this PR as I think this might be better? Let me know if you'd like me to separate the PRs or if MemoryExtensions is the way to go?

```
|     Method |  Size |      Mean |     Error |    StdDev | Ratio |
|----------- |------ |----------:|----------:|----------:|------:|
|   Original | 32768 | 17.890 us | 0.0576 us | 0.0481 us |  1.00 |
|       Linq | 32768 |  2.099 us | 0.0415 us | 0.0494 us |  0.12 |
| Vectorized | 32768 |  2.029 us | 0.0222 us | 0.0197 us |  0.11 |
```

Reply:
Nice investigating and good find about the `System.Linq` option to maybe avoid the dependency!
I recall you saying before that you were having a hard time running benchmarks on .NET Framework, are you able to confirm if you're seeing the same improvement on the .NET Framework side? I'm not sure, but I suspect you may be hitting some .NET 8 optimizations there that may not be present on the .NET Framework side. It would at least be something we'd want to confirm/rule out :)